### PR TITLE
Add interfaces to output of model.getModelDefs and getModelDict

### PR DESCRIPTION
**Pull request by jnwatson**

### CODE REVIEW

File: synapse/datamodel.py [hunk 1]
```diff
@@ -518,7 +518,8 @@ def __init__(self):
         item = s_types.NodeProp(self, 'nodeprop', info, {})
         self.addBaseType(item)
 
-        info = {'doc': 'A potentially huge/tiny number. [x] <= 170141183460469231731687 with a fractional precision of 15 decimal digits.'}
+        info = {'doc': 'A potentially huge/tiny number. [x] <= 170141183460469231731687 with a fractional '
+                       + 'precision of 15 decimal digits.'}
```
Comment (vEpiphyte) [inline synapse/datamodel.py:522]: what is this extra plus sign???

Comment (jnwatson) [reply synapse/datamodel.py:522]: Do you prefer it without the plus?

Comment (vEpiphyte) [reply synapse/datamodel.py:522]: multiline strings should be automatically concatenated together?

Comment (jnwatson) [reply synapse/datamodel.py]: ok

File: synapse/datamodel.py [hunk 2]
```diff
@@ -558,14 +559,15 @@ def getModelDefs(self):
...
+        mdef['interfaces'] = list(self.ifaces.items())
         return [('all', mdef)]
 
     def getModelDict(self):
         retn = {
             'types': {},
             'forms': {},
             'tagprops': {},
-            'interfaces': {},
+            'interfaces': list(self.ifaces.items())
```
Comment (vEpiphyte) [inline synapse/datamodel.py:570]: Why is ifaces a list and not a dictionary?

Comment (jnwatson) [reply synapse/datamodel.py]: ok. Added a small test as well.

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**